src/box/xrow_update_field.c

```diff
@@ -503,10 +504,17 @@ xrow_update_op_do_splice(struct xrow_update_op *op, const char *old)
 	}
 	assert(arg->offset <= str_len);
 
-	arg->tail_offset = arg->offset + arg->cut_length;
-	arg->tail_length = str_len - arg->tail_offset;
-	op->new_field_len = mp_sizeof_str(arg->offset + arg->paste_length +
-					  arg->tail_length);
+	xrow_update_string_rope_create(str, &fiber()->gc);
+	if (str_len > 0)
+		xrow_update_string_rope_insert(str, 0, old, str_len);
+	if (arg->cut_length > 0)
+		xrow_update_string_rope_erase(str, arg->offset,
+					      arg->cut_length);
+	if (arg->paste_length > 0)
+		xrow_update_string_rope_insert(str, arg->offset, arg->paste,
+					       arg->paste_length);
+	op->new_field_len =
+		mp_sizeof_str(xrow_update_string_rope_size(str));
```

Comment:
Doing that even when the operation is just one looks like a big overkill. Same for traversing the rope with a callback ("virtual" function call on the number of nodes). I suppose it would show some noticeable perf penalty for splice operations. But I will leave that to the core team to decide if this is fine.

Reply:
Discussed with @alyapunov Decided we should try to avoid performance degradation. We probably can do the simplest thing - use old structure until there is a second update - in that case convert it to rope.

Reply:
After optimization degradation is gone:
A = 1.57
D = 0.029